### DISCUSSION TITLE
user fora?

### DISCUSSION THREAD

**Discussion by staalmannen** [open-watcom/open-watcom-v2#33] [General]
Hi I just played a bit with trying to build ReactOS with OpenWatcom v2 (just to see how far I would get). I ran OpenWatcom and CMake under Wine under Linux (seems to work great btw) against a recent svn checkout of ReactOS. Something really cool was that I could get wmake files generated from cmake and it started building (a few things under builddir\tools got built before it stopped).

Everything is basically vanilla except a few modifications to CMakeLists.txt under reactos.

It is possible that I could come further in the build if there are some people to discuss with - specifically relating to cmake and watcom. Right now I have basically just guessed on which modifications that are needed to add support for open watcom in the reactos build.

Is there a user forum or other resource (preferrably not real-time like irc since it depends that people of similar interests are in the same time zone) where I might get some feedback/help?

**Comment by pchapin** [comment]
Have you tried the Open Watcom news groups? I believe you can access them from Google Groups (although I have trouble getting that to work for me). See: http://www.openwatcom.org/ for more information.

**Comment by jmalak** [comment]
Just a note, you could use Linux native OW tools for building reactos without Wine.
I don't know if CMake is able to generate make files for OW Linux host.

**Comment by ideafarm** [comment]
I've spent hours looking, in vain, for an active forum for Open
Watcom.  Don't waste your time.  As far as I can tell, this fork on
GitHub is the best place to connect with the handful of people who are
actively maintaining Open Watcom.  If you do continue to search and
find such a group, please let me know.

On 2/9/14, Jiří Malák notifications@github.com wrote:

> Just a note, you could use Linux native OW tools for building reactos
> without Wine.
> I don't know if CMake is able to generate make files for OW Linux host.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/open-watcom/open-watcom-v2/issues/33#issuecomment-34580989

## 

---

NOTICE:  This is not a private or confidential conversation.  All email to
and from this address resides in the public record and is subject to
publication.  The general business of IdeaFarm (tm) Operations is conducted
transparently.  Pseudo-confidential communication is possible but only by

## prior arrangement, and the request itself remains in the public record.

**Comment by staalmannen** [comment]
I initially went for a clean WINEPREFIX to reduce one level of complexity (cross-compilation, plus that a clean WINEPREFIX has the advantage of being very minimal so you got lots of control over the environment). What I was especially concerned about wrt cross compilation is that some utilities are built first that handle the further build, and those would have to be built for the host os - and I could not find how to set that up. After your suggestion, I tested native Cmake after installing the linux vatiant of open watcom v2. Unfortunately, CMake (gui) on Linux does not have the options to make Watcom wmake files, so I think the easiest for now is to play around under Wine or Windows.

**Comment by staalmannen** [comment]
A question about error

Error! E1054: Expression must be constant

I found that this should be possible to suppress by passing the -aa flag to wcc386.
When I add this to CFLAGS I do however get an error message that "aa.c can not be found"
Is there an alternative way of passing this flag or has it changed in the v2 fork?

Another thing: CMake seems insistent on setting CC and CXX to wcl386 instead of wcc386 - will this influence anything?

**Comment by jmalak** [comment]
wcl... is "compile and link" utility which call appropriate compiler or linker.
It is similar as "cl" (Microsoft) or "gcc" (GNU) command.
In OW is owcc tool which should be "compatible" with POSIX.
ANyway
wcl... -c -cc ...      is same as wcc...
wcl... -c -cc++      is same as wpp...
I checked -aa option with wcl... and it is properly passed -aa option to appropriate compiler
It looks like CMake somehow remove dash for -aa option. You can try /aa.

**Comment by staalmannen** [comment]
Thanks! I got it working. Reactos "host tools" build and I am currently fighting with the second stage - building the actual reactos (will make a patch dump to that project whenever/if it works).

One other question: is there a message compiler (like VS MC.exe) for Watcom? Right now I am using windmc from mingw32-binutils. It would however be better if there was a real watcom variant :)

**Comment by jmalak** [comment]
OW doesn't contain "message compiler".
It should be possible to implement MC if some open documentation is available.

**Comment by staalmannen** [comment]
OK good to know. I was currently trying to make a stand-alone version by isolating it from the rest of binutils, but there are lots of cross-dependencies (libbiberty, bfd). The windmc in binutils [1] is based on the Wine wmc [2], which seems cleaner. Both are probably GPL so for Open Watcom it might be needed to do a clean implementation.

1: http://sourceware.org/cgi-bin/cvsweb.cgi/src/binutils/?cvsroot=src
2: http://sourceforge.net/p/wine/code/ci/master/tree/tools/wmc/

**Comment by staalmannen** [comment]
OK please be gentle :)

This is a first attempt with a port of the Wine message compiler to Open Watcom
- still needs lots of cleanup and slimming down. I got a binary that seems to execute fin under Wine/windows... I will now attempt to use it in my reactos experiment.

https://github.com/staalmannen/owmc

**Comment by jmalak** [comment]
It is excelent as first step. I can help you with integration into OW build system if you want.

**Comment by ArmstrongJ** [comment]
Before anything gets integrated into Open Watcom, I think everyone needs to stop and consider the licensing problem.  It appears that WINE's message compiler is LGPL 2.1 or greater, which is decidedly incompatible with Open Watcom's license. A discussion on whether any LGPL software is to be introduced into the Open Watcom distribution needs to take place prior to integrating this mc.

This message compiler does have the benefit of being a standalone executable, so it isn't as problematic as, let's say, integrating LGPL code into Open Watcom's runtime libraries.  However, installers would need to have the LPGL added to the installation process in addition to the OpenWatcom license.

**Comment by pchapin** [comment]
Perhaps keeping it a stand-alone project would be best in this case. Could it be stored as another repository on GitHub under the Open Watcom organization?

**Comment by rdos314** [comment]
For me the top priority is a release of OpenWatcom that works on as many platforms as possible. This should be done before too many complex things are added to the source.

**Comment by staalmannen** [comment]
Jmalak: Thanks :) I will see if I can provide you with commit access directly to this repo, otherwise it might be even better if the "official" repo is out of my hands (see below). I believe that most "libport" and "libwine" components should be possible to replace by native C-library functions.

ArmstrongJ: I agree that licensing is serious buisness. Luckily this is a binary so it does not "infect" other components like a library would do. It might be interesting to keep it as a sepparate project for now, but if it could be another repo under the OW v2 project that would be cool :)

I have also seen that users of VS Express are suffering from lacking MC.exe, so perhaps the standalone binary message compiler should have a wider goal than only supporting OW?

Sorry if this spamming of the issue list is annoying.
As an experiment, I started a subreddit, which might form a foundation for a "user forum"
http://www.reddit.com/r/OpenWatcom/

**Comment by ArmstrongJ** [comment]
@staalmannen I like the idea of another project under the open-watcom organization.  It'd be nice to keep the Open Watcom licensing simple, at least for now.  I also like the subreddit idea, and I've already left a comment!

**Comment by rdos314** [comment]
I just built the current source-base installed it and built all the RDOS device-drivers, class library + our professional terminal, and all seems to work just fine at this point. I've also checked remote-debugging of RDOS applications, and that also seems to work just fine. I built on Windows 7 (64-bit) using a 32-bit OW from january, and using the 32-bit tools. My only complaint is that I'm now unable to install the 32-bit version on a 64-bit OS (had to use the rel directory). This is not so good when you don't have GCC or Visual C++ installed on your machine. A warning would have been more appropriate.

**Comment by jmalak** [comment]
Hi Leif. 
Thanks for your test results.

I changed installer to disable install 32-bit version on 64-bit because there were some problems with file association and with Windows start menu items install/uninstall. It is probably problem with registry.
I have now no time to fix 32-bit installer to properly handle 32-bit installation on 64-bit host.
I will return to this problem later.
As workaround I will change behaviour of 32-bit installer to only warn about potential problems on 64-bit host and enable user to continue insted of break installation.
